cmd/cli/config/auto.go

```diff
@@ -0,0 +1,143 @@
+			"can be using at one time in aggregate for all jobs (values over 100 will be rejected)")
+	flags.IntVar(&settings.JobPercentage,
+		"job-percentage",
+		defaultJobPercentage,
+		"Percentage expressed as a number from 1 to 100 representing per job amount of resource the system "+
+			"can be using at one time for a single job (values over 100 will be rejected)")
+	flags.IntVar(&settings.DefaultPercentage,
+		"default-job-percentage",
+		defaultDefaultJobPercentage,
+		"TODO I DON'T KNOW WHAT THIS DOES FIX ME")
+	flags.IntVar(&settings.QueuePercentage,
+		"queue-job-percentage",
+		defaultQueuePercentage,
+		"Percentage expressed as a number from 1 to 100 representing the total amount of resource the system "+
+			"can queu at one time in aggregate for all jobs (values over 100 are accepted)")
+	return flags
+}
+
+func (a *autoSettings) validate() error {
+	if a.TotalPercentage == 0 {
+		return fmt.Errorf("total-percentage must be greater than 0")
+	}
+	if a.TotalPercentage >= 100 {
+		return fmt.Errorf("total-percentage must be less than 100")
+	}
+	if a.JobPercentage == 0 {
+		return fmt.Errorf("job-percentage must be greater than 0")
+	}
+	if a.JobPercentage >= 100 {
+		return fmt.Errorf("job-percentage must be less than 100")
+	}
+	if a.JobPercentage > a.TotalPercentage {
+		return fmt.Errorf("job-percentage must be less than or equal to total-percentage")
+	}
+	if a.DefaultPercentage == 0 {
+		return fmt.Errorf("default-job-percentage must be greater than 0")
+	}
+	if a.DefaultPercentage >= 100 {
+		return fmt.Errorf("default-job-percentage must be less than 100")
+	}
+	if a.DefaultPercentage > a.TotalPercentage {
+		return fmt.Errorf("default-job-percentage must be less than or equal to total-percentage")
+	}
+	return nil
+}
+
+func newAutoResourceCmd() *cobra.Command {
+	settings := new(autoSettings)
+	autoCmd := &cobra.Command{
+		Use:   "auto-resources [flags]",
+		Short: "Auto set compute resources values in the config.",
+		Args:  cobra.MinimumNArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := settings.validate(); err != nil {
+				return err
+			}
+			return autoConfig(settings)
+		},
+	}
+
+	autoCmd.PersistentFlags().AddFlagSet(autoFlags(settings))
+
+	return autoCmd
+}
+
+func autoConfig(settings *autoSettings) error {
+	pp := system.NewPhysicalCapacityProvider()
+	physicalResources, err := pp.GetTotalCapacity()
+	if err != nil {
+		return fmt.Errorf("failed to calculate system physical resources: %w", err)
+	}
+
+	if err := setResources(types.NodeComputeCapacityTotalResourceLimits, settings.TotalPercentage, physicalResources); err != nil {
+		return err
+	}
+	if err := setResources(types.NodeComputeCapacityJobResourceLimits, settings.JobPercentage, physicalResources); err != nil {
+		return err
+	}
+	if err := setResources(types.NodeComputeCapacityDefaultJobResourceLimits, settings.DefaultPercentage, physicalResources); err != nil {
```

Comment:
Here is the only place we use this value: https://github.com/bacalhau-project/bacalhau/blob/main/pkg/node/compute.go#L128. The calculator then merges the `DefaultJobResourceLimits` with a tasks `ResourcesConfig`. If any value in the task (job) resource config is zero then the default is substituted in. So it serves as the default on compute nodes for jobs that don't specify resources.

`JobResourceLimits` is used in https://github.com/bacalhau-project/bacalhau/blob/main/pkg/node/compute.go#L171 and https://github.com/bacalhau-project/bacalhau/blob/main/pkg/node/compute.go#L202. It represents the max resources for a job a compute node bids on. 

Sorry if you already knew that, but writing it out was helpful for me. 

>  If this is true and we set the default to 75% aren't we forcing sequential execution

Yes, in the event we (a compute node) receive a job that doesn't specify any resources, we set its default limit to 75% of our system. For what its worth in this [code](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/node/config_compute.go#L153) we are using 80% if unspecified (but I am trying to move us away from that code because too many configs).

Do you think we should have a default of less than 75% here, maybe 45%? My worry with lower values is that it could lead to jobs take a really long time to run. On the other hand, this command is totally optional, and I am adding it to help users mitigate issues like: https://github.com/bacalhau-project/bacalhau/issues/2980

------------------------

Separately, it feels strange for jobs to be submitted to the network that don't specify resource requirements. Perhaps it would be worth making `DefaultJobResourceLimits` a value defined by the client submitting the job.

Reply:
Thanks, that's really useful.  I guess it'd be equally useful for us to have this information about resource usage for compute nodes in the docs somewhere.

> Separately, it feels strange for jobs to be submitted to the network that don't specify resource requirements. 

I agree, but it can be hard for users to know/guess what their code is likely to use.  For CPUs [this blog post](https://home.robusta.dev/blog/stop-using-cpu-limits) has some interesting advice wrt limiting access to CPU in K8s.

FWIW today, the default for docker is no limits on CPU or RAM, so it will use all of the resources available to docker (or at least make them available) unless we set values.  Wasm has a 4gb limit of RAM (by virtue of being 32bit).